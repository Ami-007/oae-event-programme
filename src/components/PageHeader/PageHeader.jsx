import "./PageHeader.scss";
import React, {Fragment} from "react";
import EventHeaderTitle from "../../components/EventHeaderTitle/EventHeaderTitle";
import EventHeaderDetails from "../../components/EventHeaderDetails/EventHeaderDetails";
import HeaderImageVideoContainer from "../../components/HeaderImageVideoContainer/HeaderImageVideoContainer";

const PageHeader = (props) => {
  const { title, subtitle, time, date, venue, location, content, isVideo } = props;
  
  const headerClass = ((time, date, venue, location, content, isVideo) == null) ? "page-header-only" : "page-header";
 
  return (
    <div className={headerClass}>
      <EventHeaderTitle title={title} subtitle={subtitle} /> 
      {headerClass === "page-header" ? <Fragment>   
      <EventHeaderDetails time={time} date={date} venue={venue} location={location} />
      <HeaderImageVideoContainer content={content} isVideo={isVideo} />   
      </Fragment> : null}       
    </div>
  );
};

export default PageHeader;
