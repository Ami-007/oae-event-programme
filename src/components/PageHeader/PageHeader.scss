@use "../../assets/sass/variables" as *;

.page-header {
  // background-color: $modern-primary;
  background-color: var(--theme-color-default);
  padding-top: 5rem;
  &--only {
    padding-bottom: 2rem;
  }
}
