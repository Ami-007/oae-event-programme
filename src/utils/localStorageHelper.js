export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
