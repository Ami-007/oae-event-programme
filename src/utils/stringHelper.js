export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
