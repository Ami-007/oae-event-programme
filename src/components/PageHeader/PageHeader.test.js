import { customRender } from "../../utils/testUtils";
import PageHeader from "./PageHeader";

test("it should render the component", () => {
  const { container } = customRender(
    <PageHeader
      title="BACH, THE UNIVERSE & EVERYTHING"
      subtitle="Can You Hear The Shape of The Drum?"
      time="11.30am"
      date="Sun 21 Nov 2021"
      venue="South Bank"
      location="London"
      content=""
      isVideo={false}
    />
  );
  expect(container).toMatchSnapshot();
});

test("it should only render the EventTitle component if no props are passed to the other two coponents", () => {
  const { container } = customRender(
    <PageHeader
      title="BACH, THE UNIVERSE & EVERYTHING"
      subtitle="Can You Hear The Shape of The Drum?"
    />
  );
  expect(container).toMatchSnapshot();
});
