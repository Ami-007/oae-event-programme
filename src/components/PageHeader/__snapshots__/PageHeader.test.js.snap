// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should only render the EventTitle component if no props are passed to the other two coponents 1`] = `
<div>
  <div
    class="page-header--only"
  >
    <div
      class="header-home"
    >
      <span
        class="header-home__subtitle"
      >
        BACH, THE UNIVERSE & EVERYTHING
      </span>
      <h1
        class="header-home__title"
      >
        Can You Hear The Shape of The Drum?
      </h1>
    </div>
  </div>
</div>
`;

exports[`it should render the component 1`] = `
<div>
  <div
    class="page-header"
  >
    <div
      class="header-home"
    >
      <span
        class="header-home__subtitle"
      >
        BACH, THE UNIVERSE & EVERYTHING
      </span>
      <h1
        class="header-home__title"
      >
        Can You Hear The Shape of The Drum?
      </h1>
    </div>
    <div
      class="event-header-details"
    >
      <p
        class="event-header-details__time-date"
      >
        11.30am
        <span
          class="event-header-details__divider"
        >
          |
        </span>
        Sun 21 Nov 2021
      </p>
      <p
        class="event-header-details__venue-location"
      >
        <span
          class="event-header-details__venue"
        >
          South Bank
        </span>
        <span
          class="event-header-details__comma"
        >
          ,
        </span>
         
        London
      </p>
    </div>
  </div>
</div>
`;
